domain/src/test/kotlin/no/nav/su/se/bakover/domain/revurdering/VurderOpphørVedRevurderingTest.kt

```diff
@@ -0,0 +1,247 @@
+        val vilkårsvurderingerMock = mock<Vilkårsvurderinger> {
+            on { resultat } doReturn Resultat.Innvilget
+        }
+        val månedsberegningMock1 = mock<Månedsberegning> {
+            on { periode } doReturn Periode.create(1.november(2021), 30.november(2021))
+            on { erSumYtelseUnderMinstebeløp() } doReturn true
+        }
+        val månedsberegningMock2 = mock<Månedsberegning> {
+            on { periode } doReturn Periode.create(1.desember(2021), 31.desember(2021))
+            on { erSumYtelseUnderMinstebeløp() } doReturn true
+        }
+        val beregningMock = mock<Beregning> {
+            on { getMånedsberegninger() } doReturn listOf(månedsberegningMock1, månedsberegningMock2)
+        }
+
+        VurderOpphørVedRevurdering(
+            vilkårsvurderinger = vilkårsvurderingerMock,
+            beregning = beregningMock,
+            clock = fixedClock,
+        ).resultat shouldBe OpphørVedRevurdering.Ja(listOf(Opphørsgrunn.SU_UNDER_MINSTEGRENSE), 1.november(2021))
+    }
+
+    @Test
+    fun `inneværende måned under minstebeløp gir opphør`() {
+        val vilkårsvurderingerMock = mock<Vilkårsvurderinger> {
+            on { resultat } doReturn Resultat.Innvilget
+        }
+        val månedsberegningMock = mock<Månedsberegning> {
+            on { periode } doReturn Periode.create(LocalDate.now(fixedClock).startOfMonth(), LocalDate.now(fixedClock).endOfMonth())
+            on { erSumYtelseUnderMinstebeløp() } doReturn true
+        }
+        val beregningMock = mock<Beregning> {
+            on { getMånedsberegninger() } doReturn listOf(månedsberegningMock)
+        }
+
+        VurderOpphørVedRevurdering(
+            vilkårsvurderinger = vilkårsvurderingerMock,
+            beregning = beregningMock,
+            clock = fixedClock,
+        ).resultat shouldBe OpphørVedRevurdering.Ja(listOf(Opphørsgrunn.SU_UNDER_MINSTEGRENSE), LocalDate.now(fixedClock).startOfMonth())
+    }
+
+    @Test
+    fun `en fremtidig måned med beløp lik 0 gir opphør`() {
+        val vilkårsvurderingerMock = mock<Vilkårsvurderinger> {
+            on { resultat } doReturn Resultat.Innvilget
+        }
+        val månedsberegningMock = mock<Månedsberegning> {
+            on { periode } doReturn Periode.create(1.desember(2021), 31.desember(2021))
+            on { erSumYtelseUnderMinstebeløp() } doReturn false
+            on { getSumYtelse() } doReturn 0
+        }
+        val beregningMock = mock<Beregning> {
+            on { getMånedsberegninger() } doReturn listOf(månedsberegningMock)
+        }
+
+        VurderOpphørVedRevurdering(
+            vilkårsvurderinger = vilkårsvurderingerMock,
+            beregning = beregningMock,
+            clock = fixedClock,
+        ).resultat shouldBe OpphørVedRevurdering.Ja(listOf(Opphørsgrunn.FOR_HØY_INNTEKT), 1.desember(2021))
+    }
+
+    @Test
+    fun `en måned i fortiden under minstebeløp gir ikke opphør`() {
```

Comment:
Så det er innafor å revurdere historiske måneder til [0,minstebeløp) ? Men de vil da ikke føre til et opphør? Men hvis det hadde vært frem i tid ville den samme månedsberegninga ført til opphør? Gitt at Stønadsperioden er aktiv fremdeles da.

Reply:
Det stemmer. Årsaken er at vi ønsker at man skal kunne gjøre den typen saksbehandling uten at hele ytelsen opphører. Revurderings-utfallet "opphør" vil stanse alt fra opphørsdatoen (også ting i framtiden for opphørsdatoen).